typhos/suite.py

```diff
@@ -767,6 +767,63 @@ def save(self):
     # Add the template to the docstring
     save.__doc__ += textwrap.indent('\n' + utils.saved_template, '\t\t')
 
+    def save_screenshot(
+        self,
+        filename: str,
+    ) -> bool:
+        """Save a screenshot of this widget to ``filename``."""
+
+        image = utils.take_widget_screenshot(self)
+        if image is None:
+            logger.warning("Failed to take screenshot")
+            return False
+
+        logger.info(
+            "Saving screenshot of suite titled '%s' to '%s'",
+            self.windowTitle(), filename,
+        )
+        image.save(filename)
+        return True
+
+    def save_device_screenshots(
+        self,
+        filename_format: str,
+    ) -> dict[str, str]:
+        """Save screenshot(s) of devices to ``filename_format``."""
+
+        filenames = {}
+        for device in self.devices:
+            display = self.get_subdisplay(device)
+
+            if hasattr(display, "to_image"):
+                image = display.to_image()
+            else:
+                # This is a fallback for if/when we don't have a TyphosDisplay
+                image = utils.take_widget_screenshot(display)
```

Comment:
Do you know how often this happens?  I was under the impression most everything was a `TyphosDeviceDisplay`

Reply:
It shouldn't _ever_ happen in the current codebase. The truth is the following:
1. I added `utils.take_widget_screenshot` first, added tests and had it all working
2. Then I shifted to "it'd be better to save each device display" 
3. Then I recalled that I had already made `to_image` on `TyphosDeviceDisplay` (a few years ago perhaps?)
4. Then I needed to justify the work from (1) so this `else` clause stayed in